src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copy = Object.assign({}, state);
+  const objArr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const key in action.extraData) {
+          copy[key] = action.extraData[key];
+        }
+        objArr.push(Object.assign({}, copy));
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          if (copy.hasOwnProperty(key)) {
+            delete copy[key];
+          }
+        }
+        objArr.push(Object.assign({}, copy));
+        break;
+      case 'clear':
+        for (const key in copy) {
+          delete copy[key];
+        }
+        objArr.push(Object.assign({}, copy));
```

Comment:
You repeat this operation 3 times in each `case`, just place it after a switch statement

Reply:
Also, it'd be easier to use `...` - spread operator here